src/coreclr/jit/optimizer.cpp

```diff
@@ -366,6 +366,17 @@ void Compiler::optUnmarkLoopBlocks(BasicBlock* begBlk, BasicBlock* endBlk)
             break;
         }
     }
+
+    JITDUMP("\n");
+
+#if FEATURE_LOOP_ALIGN
+    if (begBlk->isLoopAlign())
```

Comment:
Ah, good catch, I think I missed this earlier although there was one rare case that I handled in :
https://github.com/dotnet/runtime/blob/f958f2bf5f6be374719cb1fbdf17131551857dd4/src/coreclr/jit/morph.cpp#L16607-L16610

Reply:
I just sent out draft PR https://github.com/dotnet/runtime/pull/51313 to test how often we retain align flag on blocks that are no longer loop. Probably not because we might have hit an assert during codegen, but just want to double check.